giraph-core/src/main/java/org/apache/giraph/utils/UnsafeByteArrayOutputStream.java

```diff
@@ -36,7 +37,7 @@
  * Byte array output stream that uses Unsafe methods to serialize/deserialize
  * much faster
  */
-public class UnsafeByteArrayOutputStream extends OutputStream
+public class UnsafeByteArrayOutputStream extends Output
```

Comment:
I didn't realize kryo doesn't provide an interface but a class we have to extend. What kind of testing have you performed for this change, since it will affect everything we do? (testing both for performance and correctness)

Reply:
So far I run unit tests and snapshot tests and couple of prod jobs. I will run with more prod jobs to verify performance and correctness.

Reply:
I don't recall whether you've already done this, but can also do some microbenchmark, e.g. just write random objects and measure the difference?

Reply:
I have done some benchmarking with the additional changes that depend on this pull request ( I will send a separate PR for that). For this PR, my testing focused on correctness and making sure that there is no performance regression.